test/integration/base_integration_test.cc

```diff
@@ -221,6 +232,30 @@ void BaseIntegrationTest::setUpstreamProtocol(FakeHttpConnection::Type protocol)
           ConfigHelper::setProtocolOptions(
               *bootstrap.mutable_static_resources()->mutable_clusters(0), protocol_options);
         });
+  } else {
+    RELEASE_ASSERT(protocol == FakeHttpConnection::Type::HTTP3, "");
+    setUdpFakeUpstream(true);
+    upstream_tls_ = true;
+    config_helper_.configureUpstreamTls(false, true);
+    config_helper_.addConfigModifier(
+        [&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+          // Docker doesn't allow writing to the v6 address returned by
+          // Network::Utility::getLocalAddress.
```

Comment:
Is this a docker bug that leads us to hardcode address here?

Reply:
Yeah, this test passes locally for me without that change, but failed on CI.  I as far as I could tell the docker container was resolving a local v6 address but was cranky at sending to its non-loopback v6 address, where it was fine doing so with v4.
AFIK we have similar issues in-house with v6 docker behaving differently so I'm not terribly surprised.